packages/datx-jsonapi/src/decorateCollection.ts

```diff
@@ -175,6 +173,33 @@ export function decorateCollection(
       return read<T>(query.url, this, reqOptions).then(handleErrors);
     }
 
+    public async getAll<T extends IJsonapiModel = IJsonapiModel>(
+      type: IType | IModelConstructor<T>,
+      options?: IRequestOptions,
+    ): Promise<IGetAllResponse<T>> {
+      let response = await this.getMany(type, options);
+
+      const data: Array<T> = [];
+      const responses: Array<Response<T>> = [];
+      let lastResponse = response;
+
+      data.push(...(response.data as Array<T>));
+      responses.push(response);
+
+      while (response.next) {
+        response = await response.next();
+        responses.push(response);
+        data.push(...(response.data as Array<T>));
+        lastResponse = response;
```

Comment:
I'm curious what is more optimal?
Assingning `lastResponse` in the loop or doing this outside the loop...
```
lastResponse = responses[responses.length - 1];
```

Reply:
U are right, I will extract it outside of the loop. There is no point in reassigning it on every iteration since we only need last one.